discover-rules/src/main/scala/org/xalgorithms/discover_rules/Job.scala

```diff
@@ -8,47 +8,72 @@ import org.apache.spark.streaming.kafka.KafkaUtils
 import org.apache.spark.streaming.{Seconds, StreamingContext}
 import com.datastax.spark.connector.streaming._
 import com.datastax.spark.connector._
+import com.mongodb.spark.MongoSpark
+import com.mongodb.spark.config.ReadConfig
+import org.joda.time.DateTime
+import org.xalgorithms.discover_rules.udt._
+import com.github.nscala_time.time.Imports._
+import java.util.TimeZone
 
 
 object Job {
-  def traverse(source: UDTValue, path: Array[String], value: String): Boolean = {
-    val p = path.head
+  def filter_items(tuple: (Invoice, Effective)): Boolean = {
+    val envelope = tuple._1.envelope
+    val effective = tuple._2
+    val country = effective.country
+    val region = effective.region
+    val party = effective.party
+    val timezone = effective.timezone
+    val starts = effective.starts
+    val ends = effective.ends
+
+    return is_valid_country(envelope.country, country) &&
+      is_valid_region(envelope.region, region) &&
+      is_valid_party(envelope.party, party) &&
+      is_valid_start(envelope.period.starts.toDateTime, starts) &&
+      is_valid_end(envelope.period.ends.toDateTime, ends)
+  }
+
+  def convertToTimezone(d: DateTime, tz: String): DateTime = {
+    // TODO: Fix this
```

Comment:
Log a task for this repo in Waffle and leave it unassigned in the Scheduled lane.